Tribler/Test/Core/test_libtorrent_download_impl.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+from Tribler.Core.Libtorrent.LibtorrentDownloadImpl import LibtorrentDownloadImpl
+from Tribler.Core.SessionConfig import SessionStartupConfig
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.Utilities.twisted_thread import deferred
+from Tribler.Test.test_as_server import TestAsServer, TESTS_API_DIR
+
+
+class TestLibtorrentDownloadImpl(TestAsServer):
+
+    def setUpPreSession(self):
+        super(TestLibtorrentDownloadImpl, self).setUpPreSession()
+        self.config = SessionStartupConfig()
+        self.config.set_state_dir(self.getStateDir())
+        self.config.set_torrent_checking(False)
+        self.config.set_multicast_local_peer_discovery(False)
+        self.config.set_megacache(True)
+        self.config.set_dispersy(True)
+        self.config.set_mainline_dht(False)
+        self.config.set_torrent_collecting(True)
+        self.config.set_libtorrent(True)
+        self.config.set_dht_torrent_collecting(False)
+        self.config.set_videoplayer(False)
+        self.config.set_torrent_collecting_dir(os.path.join(self.session_base_dir, 'torrent_collecting_dir'))
+
+    def test_can_create_engine_wrapper_false(self):
+        impl = LibtorrentDownloadImpl(None, None)
+        impl.cew_scheduled = True
+        self.assertFalse(impl.can_create_engine_wrapper(), "Expected can create engine wrapper to be False but was True")
+
+    def test_get_magnet_link_none(self):
+        self.tdef = TorrentDef()
+        self.sourcefn = os.path.join(TESTS_API_DIR, 'video.avi')
+        self.tdef.add_content(self.sourcefn)
+        self.tdef.set_tracker("http://localhost/announce")
+        self.tdef.finalize()
+
+        self.torrentfn = os.path.join(self.session.get_state_dir(), "gen.torrent")
+        self.tdef.save(self.torrentfn)
+
+        impl = LibtorrentDownloadImpl(self.session, self.tdef)
+        link = impl.get_magnet_link()
+        self.assertEqual(None, link, "Magnet link was  not none while it should be!")
+
+    def test_get_tdef(self):
+        self.tdef = TorrentDef()
```

Comment:
You shouldn't be using object attributes for these vars, always use local vars. You don't need nor want stuff to persist across tests (and if you do, you are doing it wrong :D).